server/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -189,7 +189,10 @@ public void onFailure(Exception e) {
                 logger.trace("[{}] op [{}] post replication actions failed for [{}]", primary.routingEntry().shardId(), opType, request);
                 // TODO: fail shard? This will otherwise have the local / global checkpoint info lagging, or possibly have replicas
                 // go out of sync with the primary
-                finishAsFailed(e);
+                // We update the checkpoints since a refresh might fail but the operations could be safely persisted, in the case that the
+                // fsync failed the local checkpoint won't advance and the engine will be marked as failed when the next indexing operation
```

Comment:
The test seems to say that the engine just survives, which confuses me a bit, perhaps you can clarify?

Reply:
Technically the engine survives until the next operation comes in and needs to be appended into the translog since the post write operation just closes the TranslogWriter and that doesn't bubble up to the Engine.